src/OrchardCore.Modules/OrchardCore.Users/Models/RegistrationSettings.cs

```diff
@@ -0,0 +1,31 @@
+using Microsoft.Extensions.Options;
+using OrchardCore.Entities;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Users.Models
+{
+    public class RegistrationSettings
+    {
+        public bool UsersCanRegister { get; set; }
+    }
+
+    public class RegistrationSettingsConfiguration : IConfigureOptions<RegistrationSettings>
```

Comment:
You don't need an `IOptions` as everything is in the site settings. Just get the site settings where you need it (in the controller).

Reply:
Done.